src/org/folio/rest/Okapi.groovy

```diff
@@ -0,0 +1,269 @@
+package org.folio.rest
+
+import groovy.json.JsonOutput
+import org.folio.http.HttpClient
+import org.folio.utilities.Logger
+import org.folio.utilities.Tools
+
+class Okapi implements Serializable {
+    def steps
+    OkapiUser superuser = new OkapiUser()
+    OkapiTenant supertenant = new OkapiTenant(id: 'supertenant')
+    private LinkedHashMap headers = ['Content-Type': 'application/json']
+
+    private String okapiUrl = 'localhost:9130'
+    private Users users = new Users(steps, okapiUrl)
+    private Permissions permissions = new Permissions(steps, okapiUrl)
+    private Authorization auth = new Authorization(steps, okapiUrl)
+
+    private Tools tools = new Tools()
+    private HttpClient http = new HttpClient()
+    private Logger logger = new Logger(steps, this.getClass().getCanonicalName())
+
+    Okapi(steps, okapiUrl) {
+        this.steps = steps
+        this.okapiUrl = okapiUrl
+    }
+
+    void superuserLogin() {
+        if (superuser.username && superuser.password) {
+            superuser.setToken(auth.getOkapiToken(supertenant, superuser))
```

Comment:
Token can expire during job execution. We should request token on each call or at least check expiration date and renew the one.